format/applebookmark/apple_bookmark.jq

```diff
@@ -0,0 +1,41 @@
+def _apple_bookmark_torepr:
+  def _f:
+    ( if .type == "string" then .data | tovalue
+      elif .type == "data" then .data | tovalue
+      elif .type == "byte" then .data | tovalue
+      elif .type == "short" then .data | tovalue
+      elif .type == "int" then .data | tovalue
+      elif .type == "long" then .data | tovalue
+      elif .type == "float" then .data | tovalue
+      elif .type == "double" then .data | tovalue
+      elif .type == "date" then .data | tovalue
+      elif .type == "boolean_false" then false
+      elif .type == "boolean_true" then true
+      elif .type == "flag_data" then
+        ( .enabled_property_flags as $eflags 
+          | .property_flags 
+          | to_entries
+          | map(select($eflags[.key] and (.key | startswith("reserved") | not)))
+          | from_entries
```

Comment:
Indent left one? I see you get a hang of jq! nice :) btw there is a `with_entries` to do `to_entries | f | from_entries` kindof but i usually like it explicit like this

Reply:
Also if you want jq IDE support and documentation, i use https://github.com/wader/jq-lsp, https://github.com/wader/vscode-jq and https://github.com/wader/jq-dash-docset more it a lot nicer to work with jq

Reply:
Thanks, wish I had set all of that up sooner!

Reply:
👍  sorry i should have mention it earlier 😄 the syntax check and also check for undefined functions and variables is very nice to have. Note that when working with fq related jq-code it might show undefined function errors at times as the jq code in fq runs in a bit different environment (can be workaround with a `.jq-lsp.jq` file but maybe not worth it).

The install of it is a bit cumbersome, hope it was ok? I should really try to make it nicer, fix binaries for jq-lsp and add vscode-jq to the vscode package repository thingy.

Reply:
It was pretty easy. I use neovim with https://github.com/neoclide/coc.nvim and just added this to my config, it works perfectly:
```
{
  "languageserver": {
    "jq": {
      "command": "jq-lsp",
      "filetypes": ["jq"]
    }
  }
}
```
```

Reply:
Wow did not expect it to work with neovim just like that :) now i have to try it